### PR TITLE
Fixing donation page payment crash

### LINKED ISSUE #1413: Crash displaying payment chooser page
## Error in Plant-for-the-Planet App

**TypeError** in **/donate-trees**
Cannot read property 'contribution' of null

[View on Bugsnag](https://app.bugsnag.com/plant-for-the-planet/plant-for-the-planet-app/errors/5d7a54c266418300197df46f?event_id=5d7a54c2004fcd76d3f80000&i=gh&m=ci)

## Stacktrace

    http://0.0.0.0:8080/bundle.js:158011 - CheckoutForm._this.handlePayment

[View full stacktrace](https://app.bugsnag.com/plant-for-the-planet/plant-for-the-planet-app/errors/5d7a54c266418300197df46f?event_id=5d7a54c2004fcd76d3f80000&i=gh&m=ci)

*Created automatically via Bugsnag*

### LINKED ISSUE #1439: Crash displaying payment chooser page (2)
## Error in Plant-for-the-Planet App

**TypeError** in **/donate-trees**
Cannot read property '0' of undefined

[View on Bugsnag](https://app.bugsnag.com/plant-for-the-planet/plant-for-the-planet-app/errors/5d836bcff15fc80019ce0d1f?event_id=5d836bcf0050497ef7690000&i=gh&m=ci)

## Stacktrace

    http://0.0.0.0:8080/bundle.js:119537 - PaymentSelector.decorateSuccess

[View full stacktrace](https://app.bugsnag.com/plant-for-the-planet/plant-for-the-planet-app/errors/5d836bcff15fc80019ce0d1f?event_id=5d836bcf0050497ef7690000&i=gh&m=ci)

*Created automatically via Bugsnag*

### PR DESCRIPTION
**Pull request by mahmed0715**
Draft: 
Fix #1413
Fix #1439
Replace PR https://github.com/Plant-for-the-Planet-org/treecounter-app/pull/1441

### CODE REVIEW

**Comment by mahmed0715** [comment]
Tested on "test" environment
with gift trees: 
- selecting existing user
- to non exiting user
- to individual 
- to company
Works good to me.

@norbertschuler Please test

I think we need to think about putting some restrictions on component definition where we should define/pass all the required parameters thus this kind of missing params issue never happens...
Thoughts please

**Comment by mahmed0715** [comment]
@sagararyal which backend you are using? We have been testing on "test" backend. It worked good on test BE.

**Comment by mahmed0715** [comment]
Screenshot
next button works

File: app/components/DonateTrees/index.js [hunk 1]
```diff
@@ -525,6 +525,7 @@ export default class DonateTrees extends Component {
                 </div>
 
                 <div>
+                  {/* {console.log('PaymentStatus', this.props.paymentStatus)} */}
```
Comment (norbertschuler) [inline app/components/DonateTrees/index.js]: If you do not need this any more, I think this line can be deleted again.

File: app/components/Payment/PaymentSelector.js [hunk 2]
```diff
@@ -69,32 +69,42 @@ class PaymentSelector extends Component {
...
+                this.props.currentUserProfile
+              );
+            }
+            setProgressModelState(false);
+          });
+      } else {
+        setProgressModelState(false);
+        this.props.paymentFailed({
+          status: false,
+          message: 'donation id missing error'
```
Comment (norbertschuler) [inline app/components/Payment/PaymentSelector.js]: I really hope this just can not happen any more now with this patch. Therefore I have not added a translation for this error message. If we want to keep this error check (I am talking about the whole `if (donationId) {...} else {}` statement) we must add a translation for this error message. So do we want to delete the if-clause again or add a translation resource?

File: app/components/Payment/PaymentSelector.js [hunk 3]
```diff
@@ -161,6 +171,7 @@ class PaymentSelector extends Component {
     if (gatewayProps.context.supportTreecounter) {
       giftToName = gatewayProps.context.supportTreecounter.displayName;
     }
+    // console.log(paymentDetails, paymentMethods)
```
Comment (norbertschuler) [inline app/components/Payment/PaymentSelector.js]: Let's not keep commented console.log statements in the code.

File: app/components/StripePayment/CheckoutForm.js [hunk 4]
```diff
@@ -98,42 +98,50 @@ class CheckoutForm extends React.Component {
...
+                  this.props.currentUserProfile
+                );
+              }
             }
-          }
+          });
+      } else {
+        this.props.paymentFailed({
+          status: false,
+          message: 'donation id missing error'
```
Comment (norbertschuler) [inline app/components/StripePayment/CheckoutForm.js]: Same as above. Either removed the if-clause with the check for `donationId` again or add this message as a translation resource (same resource as above of course, we do not need two new resources).

File: app/containers/DonateTrees/index.js [hunk 5]
```diff
@@ -45,7 +45,9 @@ class DonationTreesContainer extends PureComponent {
     // console.log('In donate Tree Route' + this.props.navigation);
     // console.log(this.props.navigation);
   }
-
+  componentWillReceiveProps(nextProps) {
+    // console.log('changed props in donate container', nextProps)
```
Comment (norbertschuler) [inline app/containers/DonateTrees/index.js]: I won't want to keep console.log statement which we do not need. By the way. For the production release I think console.log statement will be stripped out automatically, so they do not hurt us on production.

File: app/containers/GiftTrees/index.js [hunk 6]
```diff
@@ -34,6 +34,9 @@ class GiftTreesContainer extends Component {
       this.props.fetchCurrencies();
     }
   }
+  componentWillReceiveProps(nextProps) {
+    // console.log('changed props in gift container', nextProps)
```
Comment (norbertschuler) [inline app/containers/GiftTrees/index.js]: Same as above regarding commented console.log statements.

File: app/containers/StripePayment/index.js [hunk 7]
```diff
@@ -30,6 +30,12 @@ class StripePayment extends Component {
 
   render() {
     let props = this.props;
+    const donationId = props.donationId
+      ? props.donationId
+      : props.paymentStatus
+        ? props.paymentStatus.contribution[0].id
+        : null;
+    // console.log('donationId', donationId, props.paymentStatus);
```
Comment (norbertschuler) [inline app/containers/StripePayment/index.js]: Same as above regarding commented console.log statements.

**Comment by mahmed0715** [comment]
@norbertschuler updated PR. with changes requested.

File: app/components/Payment/PaymentSelector.js [hunk 8]
```diff
@@ -207,7 +206,8 @@ class PaymentSelector extends Component {
           if ('stripe_cc' === gateway) {
             const donationId = this.props.donationId
               ? this.props.donationId
-              : this.props.paymentStatus
+              : this.props.paymentStatus &&
+                this.props.paymentStatus.contribution
```
Comment (norbertschuler) [inline app/components/Payment/PaymentSelector.js]: I will apply this patch also to the other two place where `this.props.paymentStatus.contribution[0].id` is used.

**Pull request merged**